eng/pipelines/common/templates/steps/configure-sql-server-win-step.yml

```diff
@@ -195,6 +195,44 @@ steps:
   displayName: 'Setup SQL Alias [Win]'
   condition: ${{parameters.condition }}
 
+- powershell: |
+    # Create Certificate
+    $computerDnsName = [System.Net.Dns]::Resolve($null).HostName
+    $certificate = New-SelfSignedCertificate -DnsName $computerDnsName,localhost -CertStoreLocation cert:\LocalMachine\My -FriendlyName test99 -KeySpec KeyExchange
+
+    # Get path to Private key (used later)
+    $keyPath = $certificate.PrivateKey.CspKeyContainerInfo.UniqueKeyContainerName
+    $machineKeyPath = "$env:ProgramData\Microsoft\Crypto\RSA\MachineKeys\$keyPath"
+
+    # Add certificate to trusted roots
+    $store = new-object System.Security.Cryptography.X509Certificates.X509Store(
+        [System.Security.Cryptography.X509Certificates.StoreName]::Root,
+        "localmachine"
+    )
+
+    $store.open("MaxAllowed") 
+    $store.add($certificate) 
+    $store.close()
```

Comment:
We're going to add a new certificate to the Computer and to the Trusted Root Certificate Authorities certificate stores on every CI run. Depending how often the test agents are reprovisioned, we might bump into [KB2801679](https://learn.microsoft.com/en-us/troubleshoot/windows-server/windows-security/ssltls-communication-problems-after-install-kb931125).

Another approach might be to iterate over the list of instances, then lazily generate a computer certificate if one is missing, then add the certificate to the Root store, then to add permissions over its machine key path.